synapse/replication/http/devices.py

```diff
@@ -78,5 +79,54 @@ async def _handle_request(  # type: ignore[override]
         return 200, user_devices
 
 
+class ReplicationUploadKeysForUserRestServlet(ReplicationEndpoint):
+    """Ask master to upload keys for the user and send them out over federation to
+    update other servers.
+
+    This must happen on master so that the results can be correctly cached in
+    the database and streamed to workers.( Is this accurate?)
+
+    Calls to e2e_keys_handler.upload_keys_for_user(user_id, device_id, body) on
+    master to accomplish this.
+    """
+
+    NAME = "upload_keys_for_user"
+    PATH_ARGS = ("user_id",)
```

Comment:
Below we serialize the `user_id` into the body, not the path?

It likely doesn't matter *which* we do, but probably don't need both?

Reply:
Gives it a path on servlet to differentiate the request? I think?

Reply:
Successfully removed.